sdk/tables/Azure.Data.Tables/tests/TableConnectionStringTests.cs

```diff
@@ -101,6 +106,23 @@ public void ParseFailsWithInvalidConnString(string connString)
             Assert.That(TableConnectionString.TryParse(connString, out TableConnectionString tcs), Is.False, "Parsing should not have been successful");
         }
 
+        public static IEnumerable<object[]> ValidPrimaryEndpointUris()
+        {
+            yield return new object[] { new Uri($"https://{AccountName}.table.core.windows.net/") };
+            yield return new object[] { new Uri($"https://{AccountName}.table.cosmos.azure.com:443/") };
+            yield return new object[] { new Uri($"https://127.0.0.1:10002/{AccountName}/") };
+        }
+
+        [Test]
+        [TestCaseSource(nameof(ValidPrimaryEndpointUris))]
+        public void GetSecondaryUriFromPrimary(Uri endpoint)
+        {
+            Uri secondaryEndpoint = TableConnectionString.GetSecondaryUriFromPrimary(endpoint);
+
+            Assert.That(secondaryEndpoint, Is.Not.Null.Or.Empty, "Secondary endpoint should not be null or empty");
+            Assert.That(secondaryEndpoint.AbsoluteUri.IndexOf($"{AccountName}{TableConstants.ConnectionStrings.SecondaryLocationAccountSuffix}", StringComparison.OrdinalIgnoreCase), Is.GreaterThanOrEqualTo(0), $"Secondary endpoint should contain '{AccountName}-secondary', but was {secondaryEndpoint.AbsoluteUri}");
```

Comment:
Let's have a stricter assert against a known expected value.

Reply:
Probably not a bad idea.